File: software/examples/c/example_callback.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_humidity.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

// Callback function for humidity callback (parameter has unit %RH/10)
void cb_humidity(uint16_t humidity) {
	printf("Relative Humidity: %f %%RH\n", humidity/10.0);
}

int main() {
	// Create IP connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	Humidity h;
	humidity_create(&h, UID); 

	// Add device to IP connection
	if(ipcon_add_device(&ipcon, &h) < 0) {
		fprintf(stderr, "Could not connect to Bricklet\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Set Period for humidity callback to 1s (1000ms)
	// Note: The callback is only called every second if the 
	//       humidity has changed since the last call!
	humidity_set_humidity_callback_period(&h, 1000);

	// Register humidity callback to function cb_humidity
	humidity_register_callback(&h, HUMIDITY_CALLBACK_HUMIDITY, cb_humidity);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon);
}

File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_humidity import Humidity

# Callback function for humidity callback (parameter has unit %RH/10)
def cb_humidity(rh):
    print('Relative Humidity: ' + str(rh/10.0) + ' %RH')

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    h = Humidity(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Set Period for rh callback to 1s (1000ms)
    # Note: The callback is only called every second if the 
    #       humidity has changed since the last call!
    h.set_humidity_callback_period(1000)

    # Register humidity callback to function cb_humidity
    h.register_callback(h.CALLBACK_HUMIDITY, cb_humidity)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()
